src/metrics/peer.go

```diff
@@ -95,6 +100,76 @@ func (pm *Peer) ConnectionAttemptEvent(succeed bool, err string) {
 	}
 }
 
+// Fetch Peer information from another Peer info
+func (pm *Peer) FetchPeerInfoFromPeer(newPeer Peer) {
+	// Somehow weird to update the peerID, since it is going to be the same one
+	pm.PeerId = getNonEmpty(pm.PeerId, newPeer.PeerId)
+	pm.NodeId = getNonEmpty(pm.NodeId, newPeer.NodeId)
+	// Check User Agent and derivated client type/version/OS
+	pm.UserAgent = getNonEmpty(pm.UserAgent, newPeer.UserAgent)
+	pm.ClientOS = getNonEmpty(pm.ClientOS, newPeer.ClientOS)
+	if newPeer.ClientName != "" || pm.ClientName == "" || pm.ClientName == "Unknown" {
+		pm.ClientName = newPeer.ClientName
+		pm.ClientVersion = newPeer.ClientVersion
+	}
+	pm.Pubkey = getNonEmpty(pm.Pubkey, newPeer.Pubkey)
+	pm.Addrs = getNonEmpty(pm.Addrs, newPeer.Addrs)
+	pm.Ip = getNonEmpty(pm.Ip, newPeer.Ip)
+	if pm.City == "" || pm.City == "Unknown" || newPeer.City != "" {
+		pm.City = newPeer.City
+		pm.Country = newPeer.Country
+	}
+	if newPeer.Latency > 0 {
+		pm.Latency = newPeer.Latency
+	}
+	// Metadata requested
+	if !pm.MetadataRequest {
+		pm.MetadataRequest = newPeer.MetadataRequest
+	}
+	if !pm.MetadataSucceed {
+		pm.MetadataSucceed = newPeer.MetadataSucceed
+	}
+	// Beacon Metadata and Status
+	if newPeer.BeaconMetadata != (BeaconMetadataStamped{}) {
+		pm.BeaconMetadata = newPeer.BeaconMetadata
+	}
+	if newPeer.BeaconStatus != (BeaconStatusStamped{}) {
+		pm.BeaconStatus = newPeer.BeaconStatus
+	}
+	// Aggregate connections and disconnections
+	for _, time := range newPeer.ConnectionTimes {
```

Comment:
Not sure why this. Why not just this without the loop?

```go
pm.ConnectionEvent(newPeer.ConnectedDirection, time)
```

Reply:
what if we want to make a fetch info from a peer that had more than 1 connection, this way either if there are no connections, either there is more than 1, it will include any slice of the ConnectedTimes[] to the peer where we are fetching the info.
Does this make sense?

Reply:
Yep, sorry, I wasnt realising that the input was another peer, and of course as you are saying that other peer can have also other connection times. perhaps we can call it `MergePeer` or something like that.